src/metaxy/metadata_store/base.py

```diff
@@ -1893,13 +2006,9 @@ def _resolve_update_native(
                 )
             else:
                 return Increment(
```

Comment:
Why are we collecting to polars here? I'd prefer to preserve the actual backend of the metadata store. 

Users can always convert to polars if needed.